### ISSUE TITLE
[BUG] scap-driver: compilation failure with kernel 6.2.0

### ISSUE THREAD

**Issue by hhoffstaette** [falcosecurity/libs#918]
(previous sysdig contributor & Gentoo maintainer here :)

The old-style scap driver fails to build against kernel 6.2.0:
```
  CC [M]  /tmp/portage/dev-util/scap-driver-0.29.3/work/libs-e5c53d648f3c4694385bbe488e7d47eaa36c229a_build/driver/src/main.o
/tmp/portage/dev-util/scap-driver-0.29.3/work/libs-e5c53d648f3c4694385bbe488e7d47eaa36c229a_build/driver/src/main.c: In function 'scap_init':
/tmp/portage/dev-util/scap-driver-0.29.3/work/libs-e5c53d648f3c4694385bbe488e7d47eaa36c229a_build/driver/src/main.c:2501:30: error: assignment to 'char * (*)(const struct device *, umode_t *)' {aka 'char * (*)(const struct device *, short unsigned int *)'} from incompatible pointer type 'char * (*)(struct device *, umode_t *)' {aka 'char * (*)(struct device *, short unsigned int *)'} [-Werror=incompatible-pointer-types]
 2501 |         g_ppm_class->devnode = ppm_devnode;
      |                              ^
cc1: some warnings being treated as errors
make[2]: *** [scripts/Makefile.build:252: /tmp/portage/dev-util/scap-driver-0.29.3/work/libs-e5c53d648f3c4694385bbe488e7d47eaa36c229a_build/driver/src/main.o] Error 1
make[1]: *** [Makefile:2021: /tmp/portage/dev-util/scap-driver-0.29.3/work/libs-e5c53d648f3c4694385bbe488e7d47eaa36c229a_build/driver/src] Error 2
make[1]: Leaving directory '/tmp/linux-6.2.0'
```
This cryptic message stems from the fact that apparently some attributes of the `struct device` returned from class_create() are const since 6.2 and the conditional `ppm_devnode` prototypes no longer match, since they are declared to take a non-const `struct device*`.
The fix is easy: more `#ifdef`s :laughing:

**Comment by hhoffstaette** [comment]
Here's a patch. Please note that I'm poking around in an older release, so no PR but the code in current git master looks the same:

```
holger>diff -up main.c~ main.c
--- main.c	2023-02-21 18:46:03.019270082 +0100
+++ main.c	2023-02-21 18:49:18.129190042 +0100
@@ -2347,11 +2347,15 @@ static int get_tracepoint_handles(void)
 #endif
 
 #if LINUX_VERSION_CODE > KERNEL_VERSION(2, 6, 20)
+#if LINUX_VERSION_CODE >= KERNEL_VERSION(6, 2, 0)
+static char *ppm_devnode(const struct device *dev, umode_t *mode)
+#else
 #if LINUX_VERSION_CODE >= KERNEL_VERSION(3, 3, 0)
 static char *ppm_devnode(struct device *dev, umode_t *mode)
 #else
 static char *ppm_devnode(struct device *dev, mode_t *mode)
-#endif
+#endif /* LINUX_VERSION_CODE > KERNEL_VERSION(3, 3, 0) */
+#endif /* LINUX_VERSION_CODE > KERNEL_VERSION(6, 2, 0) */
 {
 	if (mode) {
 		*mode = 0400;
```

With this fix the driver compiles on 6.2.

**Comment by FedeDP** [comment]
Hi! Thank you very much! I am going to upgrade my arch box in a couple of days (as soon as it enters core repo) and will test this!
Btw mind to open a PR?
Also, it would be great to add a cron CI job to test the build against latest kernel in arch with driverkit, like a weekly job. I will work on it tomorrow :)
